File: cashflow/src/main/java/com/backend/cashflow/verification/PasswordVerify.java
package com.backend.cashflow.verification;

import org.springframework.stereotype.Component;

@Component
public class PasswordVerify {

    public boolean isPasswordValid(String password) {
        String uppercaseRegex = "[A-Z]";
        String lowercaseRegex = "[a-z]";
        String specialCharacterRegex = "[!@#\\$%^&*()]";
        String numberRegex = "[0-9]";

        boolean containsUppercase = password.matches(".*" + uppercaseRegex + ".*");
        boolean containsLowercase = password.matches(".*" + lowercaseRegex + ".*");
        boolean containsCharacterRegex = password.matches(".*" + specialCharacterRegex + ".*");
        boolean containsNumberRegex = password.matches(".*" + numberRegex + ".*");

        return containsUppercase && containsLowercase && containsCharacterRegex && containsNumberRegex;
    }
}

File: cashflow/src/main/java/com/backend/cashflow/verification/PasswordVerify.kt
package com.backend.cashflow.verification

import org.springframework.stereotype.Component

@Component
class PasswordVerify {
    fun isPasswordValid(password: String): Boolean {
        val uppercaseRegex = Regex("[A-Z]")
        val lowercaseRegex = Regex("[a-z]")
        val specialCharacterRegex = Regex("[!@#\$%^&*()]")
        val numberRegex = Regex("[0-9]")

        val containsUppercase = uppercaseRegex.containsMatchIn(password)
        val containsLowercase = lowercaseRegex.containsMatchIn(password)
        val containsCharacterRegex = specialCharacterRegex.containsMatchIn(password)
        val containsNumberRegex = numberRegex.containsMatchIn(password)

        return containsUppercase && containsLowercase && containsCharacterRegex && containsNumberRegex
    }
}